### ISSUE TITLE
Lift errors

### ISSUE THREAD

**Issue by pitag-ha** [pitag-ha/cardio-crumble#4]
We currently have quite an ad-hoc implementation for `cardio-crumble`. In particular, it currently doesn't have a library-binary structure. One of the things we'll need to do for a hygienic structure is to avoid raising wherever we encounter errors. Instead, we should use the OCaml `Result.t` data structure in order to lift errors step by step up to the entry modules `Simple_engine` and `Stat_engine`.

One example of that is the termination of the process in `Midi.Device.play`. Instead, `Midi.Device.play` should return a `Result.t` value, which I suggest promoting via the `Play` module up to the `Simple_engine` and `Stat_engine` modules, where the `Result.t` value should be handled by exiting if there's an error.

As a first step for this, you could only lift the error from `Midi.Device.play` until the `Play` module and handle it in there already.

**Comment by AryanGodara** [comment]
@pitag-ha , Can I work on this after #3 is done?
This seems like a longer issue, to get to more familiar with Ocaml!

**Comment by pitag-ha** [comment]
Hi @AryanGodara, nice to see that you're keen on contributing to the MIDI project! I think it's better to stay focused on one task before taking on others. Do you prefer this issue or #3 to work on? I think they're similarly involved. #3 requires a little bit of knowledge about music as well.

In case you (or anyone else) is worried about issue shortage: I, and possibly @rand00 as well, will create more issues these days.

**Comment by AryanGodara** [comment]
No worries @pitag-ha , I'll start work on this first thing tomorrow morning!
This will be an interesting task, since I've to learn about error handling in ocaml anyways (I'm still going through docs are learning everything for each PR). Fun way to spend the weekend :)

**Comment by pitag-ha** [comment]
Ok, noted. No stress!

**Comment by AryanGodara** [comment]
> @AryanGodara , I've just unassigned you from this issue in case someone else wants to work on it, given that you're working on #13 now. Does that sound good to you?

Yes, thats perfect, I'll work on the CI issue for now!

**Comment by AryanGodara** [comment]
@pitag-ha Can i come back and continue with this issue now, since i've completed the other tasks?

**Comment by pitag-ha** [comment]
Hi @AryanGodara, that's up to you! I've just written a longer message about that [here](https://github.com/pitag-ha/cardio-crumble/pull/22#issuecomment-1493434838).

**Comment by pitag-ha** [comment]
I've also just seen that you're working on [an issue on `fry`](https://github.com/rand00/fry/issues/1), aren't you? Working only on one issue at a time vs working on two issues at once is now up to you as well. However, I still recommend focussing on one thing at a time.

**Comment by AryanGodara** [comment]
> I've also just seen that you're working on [an issue on `fry`](https://github.com/rand00/fry/issues/1), aren't you? Working only on one issue at a time vs working on two issues at once is now up to you as well. However, I still recommend focussing on one thing at a time.

Yes, I'll complete the fry task first. And then switch over to this one. I just did not want to leave it as it was different from the other tasks (error handling) and I started it earlier 😅
